packages/@sanity/components/src/fileinput/DropZone.js

```diff
@@ -1,43 +1,143 @@
 import React, {PropTypes} from 'react'
 import DropZone from 'react-dropzone'
+import {get, includes} from 'lodash'
 
 import styles from 'part:@sanity/components/fileinput/dropzone-style'
 
 export default class Dropzone extends React.PureComponent {
   static propTypes = {
     className: PropTypes.string,
-    multiple: PropTypes.bool
+    multiple: PropTypes.bool,
+    icon: PropTypes.func,
+    ghost: PropTypes.bool,
+    accept: PropTypes.string
   }
 
   static defaultProps = {
     multiple: true
   }
 
+  state = {
+    hasUnAcceptedFiles: false,
+    toManyFiles: false
+  }
+
+  handleDragOver = event => {
+
+    const {accept} = this.props
+
+    // Todo handle image/*
+    if (!accept || accept == 'image/*') {
+      this.setState({
+        hasUnAcceptedFiles: false
+      })
+      return
+    }
+
+    const items = get(event, 'nativeEvent.dataTransfer.items')
+
+    const mimeTypes = this.props.accept.split(',') || []
+
+    if (items) {
+      if (items.length > 1 && !this.props.multiple) {
+        this.setState({
+          toManyFiles: true
+        })
+        return
+      }
+
+      for (let i = 0; i < items.length; i++) {
+        this.setState({
+          hasUnAcceptedFiles: !includes(mimeTypes, items[i].type)
```

Comment:
Also, if the `accept` prop is supposed to be a stand in for the `accept` attribute on native file inputs, this check is too simple. Could we use this instead? https://github.com/okonet/attr-accept